src/execution/operator/csv_scanner/scanner/string_value_scanner.cpp

```diff
@@ -533,14 +536,19 @@ void StringValueResult::AddQuotedValue(StringValueResult &result, const idx_t bu
 }
 
 void StringValueResult::AddValue(StringValueResult &result, const idx_t buffer_pos) {
+	idx_t extra_delimiter_bytes =
+	    result.state_machine.dialect_options.state_machine_options.delimiter.GetValue().size() - 1;
 	if (result.last_position.buffer_pos > buffer_pos) {
 		return;
 	}
 	if (result.quoted) {
-		StringValueResult::AddQuotedValue(result, buffer_pos);
+		AddQuotedValue(result, buffer_pos - extra_delimiter_bytes);
 	} else {
-		result.AddValueToVector(result.buffer_ptr + result.last_position.buffer_pos,
-		                        buffer_pos - result.last_position.buffer_pos);
+		idx_t size = buffer_pos - result.last_position.buffer_pos - extra_delimiter_bytes;
+		if (buffer_pos < result.last_position.buffer_pos + extra_delimiter_bytes) {
+			size = 0;
```

Comment:
Is this even possible? Maybe this should be an internal exception

Reply:
Ah, I figured out why I added these for.

Is when the data ends on a delimiter
e.g.,

delim = 'ABAC'

```csv
AABAC
```

so the result would be A|NULL.
In these cases buffer_pos = 4,  result.last_position.buffer_pos = 4, extra_delim_bytes = 3.

I'm commenting this a bit better and changing the logic a bit to make the intention more clear.